File: ui/src/types/Exercise.ts
export interface Exercise {
    id: number;
    name: string;
    description: string | null;
    muscle_group: string;
    equipment: string | null;
    instructions: string | null;
    created_at: string;
    updated_at: string;
}

export interface ExerciseCreate {
    name: string;
    description?: string | null;
    muscle_group: string;
    equipment?: string | null;
    instructions?: string | null;
}

export interface ExerciseUpdate {
    name?: string | null;
    description?: string | null;
    muscle_group?: string | null;
    equipment?: string | null;
    instructions?: string | null;
}

File: api/src/models/exercise.py
from enum import Enum

from pydantic import BaseModel, Field


class MuscleGroup(str, Enum):
    back = "back"
    chest = "chest"
    biceps = "biceps"
    triceps = "triceps"
    shoulders = "shoulders"
    legs = "legs"


class ExerciseResponse(BaseModel):
    id: int
    name: str
    description: str | None
    muscle_group: MuscleGroup
    equipment: str | None
    instructions: str | None
    created_at: str
    updated_at: str


class CreateExerciseRequest(BaseModel):
    name: str = Field(min_length=3, max_length=50)
    description: str | None = None
    muscle_group: MuscleGroup
    equipment: str | None = None
    instructions: str | None = None


class UpdateExerciseRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    muscle_group: MuscleGroup | None = None
    equipment: str | None = None
    instructions: str | None = None
